### PR TITLE
Fix `cf.aggregate` for actual_range and np.array properties

### LINKED ISSUE #764: Aggregate crashes when fields contain the actual_range attribute
I have been trying to load some monthly files containing air temperature data (from chess-met https://catalogue.ceh.ac.uk/documents/835a50df-e74f-4bfb-b593-804fd61d5eab; although the data themselves are not important for this error) using cf.load, with the intention of aggregating along the time axis. I get a crash from cf.aggregate when the files contain the `actual_range `attribute AND there are at least three files to aggregate.

This occurs because unlike some other attributes `actual_range `is not handled specially by cf.aggregate. The default behviour is to convert both to strings and append one to the other: https://github.com/NCAS-CMS/cf-python/blob/7d78eacf31202b7522e93b64e01fc20b0b50827e/cf/aggregate.py#L4871-L4881 
When the third file is appended, the string comparison is performed between a two-element np.ndarray from the new file and a string from the previous aggregation https://github.com/NCAS-CMS/cf-python/blob/7d78eacf31202b7522e93b64e01fc20b0b50827e/cf/aggregate.py#L4864-L4869. This raises an exception as an if statement requires a scalar logical (i.e. any or all). 

Since the valid_range attribute is defined in the CF-standard (https://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/build/ch02s05.html.) this should be handled. A simple correction would be to check if the property being updated is actual_range, and if so to set the parent's corresponding attribute to the minima and maxima of the two aggregated datasets. For instance:

```
        if prop in ("actual_range"):
            parent0.set_property(prop,[min(value0[0],value1[0]),max(value0[1],value1[1])])
            continue 
```

Alternatively it could be fixed by ignoring this attribute, as is currently done for valid_range & similar attributes.

### PR DESCRIPTION
**Pull request by davidhassell**
Fixes #764 

* Fix bug in `cf.aggregate` that caused a failure when properties are `numpy` arrays with two or more elements
* Fix bug in `cf.aggregate` that didn't correctly handle the "actual_range" CF attribute

#### Note
The `_equals` method can cope with string-valued arguments, so we can simplify the block of code in which it appears

### CODE REVIEW

File: cf/aggregate.py [hunk 1]
```diff
@@ -4852,20 +4852,36 @@ def _aggregate_2_fields(
...
+                # value0 and value1
+                actual_range = (
+                    min(value0[0], value1[0]),
+                    max(value0[1], value1[1]),
+                )
+            except (TypeError, IndexError):
+                # value0 and/or value1 is not set, or is
+                # non-CF-compliant.
+                parent0.del_property(prop, None)
+            else:
```
Comment (sadielbartholomew) [inline cf/aggregate.py:4876]: You are really championing the `try`/`except`/`else` block at the moment. Is it the latest craze in Python programming (I don't read the right magazines and such)? :laughing:

**Review by sadielbartholomew — APPROVED** [review]
Fixes the issue at hand, with good coverage from new testing to expose the bugs (added tests fail on `main` in the same manner as the bug report, which is always a good sign). I have minor comments but happy overall so merge when ready.

File: cf/aggregate.py [hunk 2]
```diff
@@ -4852,20 +4852,36 @@ def _aggregate_2_fields(
...
-        if prop in ("_FillValue", "missing_value"):
+        if prop == "actual_range":
+            try:
+                # Try to extend the actual range to encompass both
+                # value0 and value1
+                actual_range = (
+                    min(value0[0], value1[0]),
+                    max(value0[1], value1[1]),
+                )
+            except (TypeError, IndexError):
```
[the remark this answers is not in the document]
Comment (davidhassell) [reply cf/aggregate.py:4872]: Good catch.

File: cf/test/test_aggregate.py [hunk 3]
```diff
@@ -664,6 +666,79 @@ def test_aggregate_trajectory(self):
             )
         )
 
+    def test_aggregate_actual_range(self):
+        """Test aggregation of actual_range"""
+        import cf
+
+        f = cf.example_field(0)
```
[the remark this answers is not in the document]
Comment (davidhassell) [reply cf/test/test_aggregate.py:673]: Thanks - that was a debugging convenience that I forgot to remove :)

File: cf/aggregate.py [hunk 4]
```diff
@@ -4876,9 +4892,24 @@ def _aggregate_2_fields(
...
+    actual_range = parent0.get_property("actual_range", None)
+    if actual_range is not None:
+        valid_range = parent0.get_property("valid_range", None)
+        if valid_range is not None:
+            try:
+                if (
+                    actual_range[0] < valid_range[0]
+                    or actual_range[1] > valid_range[1]
+                ):
+                    parent0.del_property("actual_range", None)
```
[the remark this answers is not in the document]
Comment (davidhassell) [reply cf/aggregate.py:4909]: I'll but this one :)

[the remark this answers is not in the document]
Comment (davidhassell) [reply cf/aggregate.py]: ... plus a bit more wrapping: 73b8647b587aeb6f3d13f636c06d59e27f6f076d

File: cf/aggregate.py (hunk 1 above)
Comment (davidhassell) [reply cf/aggregate.py:4876]: Ha ha! "latest" doesn't really describe me, I feel :). It's just good practice, right?

**Pull request merged**